### PR TITLE
docs: Add BGP GSG

### PR DESCRIPTION
**Pull request by christarazi**
Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

File: Documentation/gettingstarted/bgp.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Specifically, if a ``Service`` of type ``LoadBalancer`` is created, Cilium will
+allocate an IP for it from a specified pool. Once the IP is allocated, the
+Agents will announce via BGP depending on the ``Service``'s
+``ExternalTrafficPolicy``. See MetalLB's `documentation
+<https://metallb.universe.tf/usage/#bgp>`_ on this specific topic.
+
+.. include:: ../beta.rst
+
+Deploy Cilium
+===============
```
Comment (qmonnet) [inline Documentation/gettingstarted/bgp.rst]:
```suggestion
=============
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Formatting nits: Could you please make sure you use the same level of indentation for all your blocks? Ideally left-align on the name of the directive (`code-block`), like you did for the `parsed-literal` with 3 spaces.

Looks all good otherwise.

File: Documentation/gettingstarted/bgp.rst (hunk 1 above)
Comment (christarazi) [reply Documentation/gettingstarted/bgp.rst]: Thanks, fixed. Even with the align column in Vim, I still missed it :sweat_smile:

Comment (qmonnet) [reply Documentation/gettingstarted/bgp.rst]: Just don't tell that to the Goland aficionados :)

**Review by qmonnet — APPROVED** [review]

File: Documentation/gettingstarted/bgp.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+       matchLabels:
+         svc: test-lb
+     template:
+       metadata:
+         labels:
+           svc: test-lb
+       spec:
+         containers:
+         - name: web
+           image: nginx
```
Comment (errordeveloper) [inline Documentation/gettingstarted/bgp.rst:110]: It looks like nginx is only used as example app in other page, do we have a demo app that is good for this? If not, I think [podinfo](https://github.com/stefanprodan/podinfo) could be a good candidate. Just a thought, and I guess here it doesn't matter all that much as we are not doing blue/green deployments or anything like that, but may a simple app that has Cilium logo would be nice.

**Review by errordeveloper — APPROVED** [review]
This looks good overall, just added a comment with some thoughts regarding the demo app.

**Pull request merged**